python/strings_udf/strings_udf/_lib/tables.pyx

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from libc.stdint cimport uint8_t, uintptr_t
+
+from strings_udf._lib.cpp.strings_udf cimport (
+    get_character_flags_table as cpp_get_character_flags_table,
+)
+
+import numpy as np
+
+
+def get_character_flags_table_ptr():
```

Comment:
I'm not quite sure why we need to plumb this all the way up to Python when it looks like the only use case is in `lowering.py` where we tell numba to forward this parameter all the way back down to the functions defined in `char_types.cuh`. Couldn't we just call the C++ function `get_character_flags_table` directly in `char_types.cuh`? Is it important to have that be a function parameter? Even if it is, couldn't we make it an optional parameter with a default and save ourselves the Cython/Python trouble?

Reply:
If I understand your question:
The `get_character_flags_table` needs to be called before the kernel launch in CPU code and not per thread in device code.